### ISSUE TITLE
Allow token parameter to be overridden on a per method basis

### ISSUE THREAD

**Issue by AnimaMundi** [slackapi/node-slack-sdk#325]
* [x] I've read and understood the [Contributing guidelines](./CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](./CODE_OF_CONDUCT.md).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

#### Description
I should be able to provide a token as a parameter for API endpoints that require a token to be sent. For example, the files.upload endpoint takes a token representing the user to upload the file as. Instead, this SDK takes the token that I initialised my WebClient with and sends that as the token parameter instead.

I would like to be able to create my WebClient using my app token, then when I make a call to upload a file, be able to provide the user token to upload the file, including bots. As of right now, I need to make a custom request directly to the API without using this SDK to do what I want.

#### Reproducible in:
node-slack-sdk version: 3.8.1
OS version(s): n/a
Device(s): n/a

#### Steps to reproduce:
1. Initialize a web client using your app token 
2. Make a call using the files.upload method, providing a user token as the value for the "token" parameter

### Expected result:
The provided user token should be sent as the value of the token parameter in JSON post body

### Actual result:
The original app token is sent as the value of the token parameter in JSON post body

**Comment by fnky** [comment]
This would be very useful for cases where we might get tokens asynchronously. Instead of having to create a new `WebClient` every time. For example, if we'd like to retrieve the token from a database and do the call.

```js
const web = new WebClient(<default token>)
const listUsers = (teamId) =>
  getTeam(teamId)
    .then(team => web.users.list({ token: team.accessToken }))
```

**Comment by aoberoi** [comment]
i agree, i think this would be a great feature to support. if anyone is willing to contribute, i'd be happy to merge it in.

**Comment by AnimaMundi** [comment]
Will take a look at this tonight, as far as I remember it's a fairly simple fix.

**Comment by aoberoi** [comment]
turns out this is a duplicate of #237, which i closed in favor of #381. so i'll be closing this issue as well.
